src/protocol/vm/state.rs

```diff
@@ -754,25 +876,103 @@ mod tests {
             .is_err());
     }
 
-    fn dai() -> ERC20Token {
-        ERC20Token::new("0x6b175474e89094c44da98b954eedeac495271d0f", 18, "DAI", U256::from(10_000))
+    #[tokio::test]
+    async fn test_get_amount_out() -> Result<(), Box<dyn std::error::Error>> {
+        setup_db("src/protocol/vm/assets/balancer_contract_storage_block_20463609.json".as_ref())
+            .await?;
+
+        let pool_state = setup_pool_state().await;
+
+        let result = pool_state
+            .get_amount_out(
+                pool_state.tokens[0],
+                U256::from_dec_str("1000000000000000000").unwrap(),
+                pool_state.tokens[1],
+            )
+            .await
+            .unwrap();
+        let new_state = result
+            .new_state
+            .as_any()
+            .downcast_ref::<VMPoolState<PreCachedDB>>()
+            .unwrap();
+        assert_eq!(result.amount, U256::from_dec_str("137780051463393923").unwrap());
+        assert_eq!(result.gas, U256::from_dec_str("102770").unwrap());
+        assert_ne!(new_state.spot_prices, pool_state.spot_prices);
```

Comment:
Can we also add checks that the block lasting overwrites are unchanged for the original pool?